### DISCUSSION TITLE
cannot find glib-object.h   vdostream sample app

### DISCUSSION THREAD

**Discussion by garethem** [AxisCommunications/discussions#217] [ACAP] [answered]
Attempting to compile an acap-native-sdk app based on the vdostream example.
Compilation in Docker fails with:

#8 0.651 In file included from siteeye_dowork.c:22:
#8 0.651 /opt/axis/acapsdk/sysroots/armv7hf/usr/include/vdo-map.h:12:10: fatal error: glib-object.h: No such file or directory
#8 0.651    12 | #include <glib-object.h>

compilation is ok for the "hello world" sample that presumably doesnt use glib-object.h

Is glib missing from the acap-native-sdk docker image?  Or am I missing a step?

**Comment by daniel-falk** [comment, marked as answer]
@garethem Try adding `gobject-2.0` to the `pkgconfig` command like this:
```diff
diff --git a/vdostream/app/Makefile b/vdostream/app/Makefile
index 0bc05dd..81839fe 100644
--- a/vdostream/app/Makefile
+++ b/vdostream/app/Makefile
@@ -2,7 +2,7 @@ PROG1   = vdoencodeclient
 OBJS1  = $(PROG1).c
 PROGS  = $(PROG1)
 
-PKGS = gio-2.0 vdostream gio-unix-2.0
+PKGS = gio-2.0 vdostream gio-unix-2.0 gobject-2.0
 
 CFLAGS += $(shell PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) pkg-config --cflags $(PKGS))
 LDLIBS += $(shell PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) pkg-config --libs $(PKGS))
 ```
 
Are you using the unmodified example? The version on latest `main` branch works for me without the change (since `gio-2.0` depends on the same include dir).
